keywords/utils.py

```diff
@@ -323,7 +323,10 @@ def compare_generic_types(object1, object2):
     elif isinstance(object1, int) and isinstance(object2, long):
         return object1 == int(object2)
     elif isinstance(object1, float) and isinstance(object2, long):
-        return object1 == float(object2)
+        if isPredictiveResult:
```

Comment:
I have no idea what are you trying to do when isPredictiveResult flag is true and why? The logic is not intuitive as the function name doesn't suggest anything regarding comparison for Predictive results. Everywhere else, we are comparing for equality, but in predictive result we are comparing for less than, that to be of difference.

I'd suggest that you either add a summary in the function doc string or at least add comment around if comparison for predictive result.

Reply:
I have added comments to the function definition and where the function is been called. Hope this makes the code more maintainable